libr/bin/format/elf/elf.c

```diff
@@ -2706,42 +2708,43 @@ static size_t get_next_not_analysed_offset(ELFOBJ *bin, size_t section_vaddr, si
 	return offset;
 }
 
-static size_t populate_relocs_record_from_section(ELFOBJ *bin, RBinElfReloc *relocs, size_t pos, size_t num_relocs) {
-	size_t size, i, j;
-	Elf_(Xword) rel_mode;
+static void populate_relocs_record_from_section(ELFOBJ *bin, RBinElfReloc *relocs, size_t num_relocs, size_t *pos) {
+	size_t p = *pos;
 
 	if (!bin->g_sections) {
-		return pos;
+		return;
 	}
 
-	for (i = 0; !bin->g_sections[i].last; i++) {
-		rel_mode = get_section_mode (bin, i);
-
-		if (!is_reloc_section (rel_mode) || bin->g_sections[i].size > bin->size || bin->g_sections[i].offset > bin->size) {
+	size_t si = 0; // section index
```

Comment:
i don't see why this change. we use `i` everywhere and here there doesn't even seem to be confusion with other loops.

Reply:
what do you mean?